components/condicionalComponents/DoctorTable.tsx

```diff
@@ -1,55 +1,62 @@
-import axios from "axios";
 import { useState } from "react";
 import { useQuery } from "react-query";
-import { Status } from "../../interfaces";
+import { Status, Submission } from "../../interfaces";
 import { useAuthStore } from "../../src/store/auth";
 import { SubmissionsTable } from "../submissions";
 import { NoContent } from "../ui";
+import {
+  getOwnSubmissions,
+  getSubmissions,
+} from "../../services/SubmissionService";
+import { Pagination } from "../../interfaces/submission";
 
-const getData = async (
-  currentPage: number,
-  status: Status | "",
-  token: string
-) => {
-  const res = await axios.get(
-    "http://localhost/api/submission?page=" + currentPage + "&status=" + status,
+export const DoctorTable = () => {
+  const [submissions, setSubmissions] = useState<Submission[]>([]);
+  const [pagination, setPagination] = useState<Pagination | null>(null);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [status, setStatus] = useState<Status | "">("");
+  const [viewOwnSubmissions, setViewOwnSubmissions] = useState(false);
+  const { user } = useAuthStore();
+
+  const { isLoading: loadingAll } = useQuery(
```

Comment:
I see, please extract these two useQuery calls into their own hook along with the useState for submissions